clustered/client/src/main/java/org/ehcache/clustered/client/config/ClusteringServiceConfiguration.java

```diff
@@ -142,6 +148,30 @@ public ClusteringServiceConfiguration(URI clusterUri, Timeouts timeouts, boolean
    */
   public ClusteringServiceConfiguration(URI clusterUri, Timeouts timeouts, boolean autoCreate, ServerSideConfiguration serverConfig, Properties properties) {
     this.clusterUri = Objects.requireNonNull(clusterUri, "Cluster URI cannot be null");
+    this.servers = null;
+    this.cacheMangerName = null;
+    this.autoCreate = autoCreate;
+    this.serverConfiguration = serverConfig;
+    this.timeouts = Objects.requireNonNull(timeouts, "Operation timeouts cannot be null");
+    this.properties = (Properties) Objects.requireNonNull(properties, "Properties cannot be null").clone();
+  }
+
+  /**
+   * Creates a {@code ClusteringServiceConfiguration} from the properties provided.
+   *
+   * @param servers the non-{@code null} Iterable<InetSocketAddress> identifying the servers in the cluster
+   * @param cacheManagerName the non-{@code null} cacheManagerName identifying the cache manager
+   * @param timeouts the {@link Timeouts} specifying the time limit for clustered cache operations
+   * @param autoCreate {@code true} if server components should be auto created
+   * @param serverConfig  the server side entity configuration required
+   *
+   * @throws NullPointerException if {@code servers} is {@code null}
+   */
+  public ClusteringServiceConfiguration(Iterable<InetSocketAddress> servers, String cacheManagerName, Timeouts timeouts,
+                                        boolean autoCreate, ServerSideConfiguration serverConfig, Properties properties) {
+    this.clusterUri = null;
+    this.servers = StreamSupport.stream(Objects.requireNonNull(servers, "Servers cannot be null").spliterator(), false).collect(toList());
```

Comment:
This seems a bit profligate... and the stream construction around a spliterator seems a bit unusual.

Reply:
Changed